### PR TITLE
fix warnings. fix issue in posix_disk_io. add python binding for info_hash_t

### PR DESCRIPTION
**Pull request by arvidn**
There are still some failures in the python binding test, but I think this fixes most of the CI failures

### CODE REVIEW

File: bindings/python/src/session.cpp [hunk 1]
```diff
@@ -248,8 +248,16 @@ namespace
...
-                p.info_hash = sha1_hash(
-                bytes(extract<bytes>(value)).arr.data());
+                if (boost::python::len(value) == 40)
+                {
+                    p.info_hash = info_hash_t(sha1_hash(
+                            bytes(extract<bytes>(value)).arr.data()));
+                }
+                else
+                {
+                    p.info_hash = info_hash_t(sha256_hash(
```
Comment (arvidn) [inline bindings/python/src/session.cpp:258]: maybe there should be a runtime check here, that the buffer is of the right size here too.

File: bindings/python/test.py [hunk 2]
```diff
@@ -230,7 +230,7 @@ def test_read_resume_data(self):
         tp = lt.read_resume_data(resume_data)
 
         self.assertEqual(tp.name, 'test')
-        self.assertEqual(tp.info_hash, lt.sha1_hash('abababababababababab'))
+        self.assertEqual(tp.info_hash.v1, lt.sha1_hash('abababababababababab'))
```
Comment (arvidn) [inline bindings/python/test.py:233]: I'm a little bit worried about this API break in python. Is there an easy way to do implicit conversions in python?

File: include/libtorrent/create_torrent.hpp [hunk 3]
```diff
@@ -101,11 +101,11 @@ namespace libtorrent {
 	// .torrent file using bencode().
 	struct TORRENT_EXPORT create_torrent
 	{
+#if TORRENT_ABI_VERSION <= 2
```
Comment (arvidn) [inline include/libtorrent/create_torrent.hpp:104]: I moved this so the document generation tool will ignore the comment

File: src/file_storage.cpp [hunk 4]
```diff
@@ -1218,7 +1218,6 @@ namespace {
 				new_mtime.push_back(0);
 
 			auto& file = new_files.back();
-			TORRENT_ASSERT(static_cast<std::int64_t>(file.size) <= max_file_size);
```
Comment (arvidn) [inline src/file_storage.cpp:1221]: this assert didn't make sense, since `file.size` is already a 48 bit bitfield, this check is always true

File: src/hasher.cpp [hunk 5]
```diff
@@ -213,7 +213,8 @@ TORRENT_CRYPTO_NAMESPACE
 #elif TORRENT_USE_CRYPTOAPI
 		m_context.update(data);
 #elif defined TORRENT_USE_LIBCRYPTO
-		SHA256_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data()), data.size());
+		SHA256_Update(&m_context, reinterpret_cast<unsigned char const*>(data.data())
+			, static_cast<std::size_t>(data.size()));
```
Comment (arvidn) [inline src/hasher.cpp:217]: I'm not sure why this didn't warn before. I'm tempted to put this in master as a separate commit, ahead of v2

File: src/posix_disk_io.cpp [hunk 6]
```diff
@@ -186,7 +186,7 @@ namespace {
 			posix_storage* st = m_torrents[storage].get();
 
 			int const piece_size = st->files().piece_size(piece);
-			int const blocks_in_piece = (piece_size + default_block_size - 1) / default_block_size;
+			int const blocks_in_piece =  st->files().blocks_in_piece2(piece);
```
Comment (arvidn) [inline src/posix_disk_io.cpp:189]: this was an old bug. I'm tempted to put this in master as a separate commit too, ahead of v2

Comment (ssiloti) [reply src/posix_disk_io.cpp:189]: `blocks_in_piece2` should only be used to compute v2 hashes, the v1 hash needs to use the v1 piece size. This function should do what `disk_io_thread::do_hash` does.

Comment (arvidn) [reply src/posix_disk_io.cpp:189]: the problem was(/is) that `block_hashes` past in here has size 64, but the previous way of computing `blocks_in_pieces` was 128, meaning we overran the end of the span in the loop below. Perhaps this is not the right fix, I copied the logic from the call site that allocates the storage for the span
